protected/humhub/libs/BasePermission.php

```diff
@@ -142,15 +143,65 @@ public function getModuleId()
      */
     public function getDefaultState($groupId)
     {
-        $configuredState = $this->getConfiguredState($groupId);
+        $defaultStoredState = $this->getDefaultStoredState($groupId);
+        if ($defaultStoredState !== null) {
+            return $defaultStoredState;
+        }
 
+        $configuredState = $this->getConfiguredState($groupId);
         if ($configuredState !== null) {
             return $configuredState;
-        } elseif ($this->defaultState == self::STATE_ALLOW) {
+        }
+
+        if ($this->defaultState == self::STATE_ALLOW) {
             return self::STATE_ALLOW;
-        } else {
-            return (int) (in_array($groupId, $this->defaultAllowedGroups));
         }
+
+        return (int) (in_array($groupId, $this->defaultAllowedGroups));
+    }
+
+    /**
+     * Returns the default state stored in DB per container type.
+     * This method returns null in case the default state for this permission or group is not stored in DB yet.
+     *
+     * @param int $groupId
+     * @return int|null
+     * @since 1.8
+     */
+    protected function getDefaultStoredState($groupId)
```

Comment:
Currently the BasePermission does not have any query logic and the permissionmanager is responsible for querying, this seperation is broken by this PR. I think the permissionManager should add this default state check instead.

Reply:
Also note that `BasePermssion` is currently used for container as well as group permission and `BasePermission::getDefaultStoredState()` implements concrete logic for container permission default state. But group level permissions also have a default state. So this logic should rather be part of `ContentContainerPermissionManager` which is responsbile for content container permission specific logic.

Reply:
ok I will review the code to move the query logic from `BasePermission` to `ContentContainerDefaultPermissionManager`.

Reply:
done in commit https://github.com/humhub/humhub/pull/4678/commits/0d6ad4c2f3241d020ff017650e8d802c5ca74def